ruby-challenges.rb

```diff
@@ -11,22 +15,143 @@
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
+us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+# p us_states.values
+# [["Washington", "Oregon", "Idaho"], ["California", "Arizona", "Nevada"], ["Maine", "New Hampshire", "Vermont"]]
+
+# p us_states.values.flatten
+# ["Washington", "Oregon", "Idaho", "California", "Arizona", "Nevada", "Maine", "New Hampshire", "Vermont"]
+
+
+# p us_states.values.flatten.sort
+# ["Arizona", "California", "Idaho", "Maine", "Nevada", "New Hampshire", "Oregon", "Vermont", "Washington"]
+
+def state_sort hash
+     hash.values.flatten.sort 
+ end
+ 
+ p state_sort(us_states)
+
+# I know .flatten is needed to eliminate the nested array issue, but I'm not sure where it goes yet. 
+
+# new_array = [us_states.map {|x| x.values_at}]
+# prints nothing in terminal
+
+# new_array = [us_states.map {|x| x.values}]
+# "prints: 43:in `block in <main>': undefined method `values' for [:northwest, ["Washington", "Oregon", "Idaho"]]:Array (NoMethodError)
+# Did you mean?  values_at"
+
+# p us_states.map {|x| x.values_at}
+# prints: '[[], [], []]'
+
+# p us_states.map {|x| x.values}
+# prints: '43:in `block in <main>': undefined method `values''
+
+# states_values = (us_states.map {|states| [states["northwest"], states["southwest"], states["northeast"]]})
+# p states_values
+# prints: ':43:in `[]': no implicit conversion of String into Integer'
+
+# puts "Array flatten implementation."
+# # print us_states.flatten
+# prints: '[:northwest, ["Washington", "Oregon", "Idaho"], :southwest, ["California", "Arizona", "Nevada"], :northeast, ["Maine", "New Hampshire", "Vermont"]]%  '
+
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+# Class Bike
+#      def initialize(model, wheels, speed)
+#          @model = model
+#          @wheels = wheels
+#          @speed = speed
+
+#          puts "The #@model bike has #@wheels and is going #@speed mph"
+#      end
+# end
 
+# trek = Bike.new("Trek", "2", "0")
+
+class Bike
+     def initialize(model)
+         @model = model
+         @wheels = 2
+         @speed = 0 
+     end
+         
+     def bike_info 
+        puts "The #@model bike has #@wheels wheels and is going #@speed mph."
```

Comment:
This is so wild how helpful Ruby is. Best practices for formatting:
string interpolation should be written as such
```ruby
def bike_info
     puts "The #{@model} bike has #{@wheels} wheels and is going #@speed mph."
end
```
However, as you can see your logic works. 
```
According to The Ruby Programming Language by Flanagan and Matsumoto:

"When the expression to be interpolated into the string literal is simply a reference to a global, instance or class variable, then the curly braces may be omitted."
```